### ISSUE TITLE
Annotation processors not run in compileTestJava

### ISSUE THREAD

**Issue by ben-manes** [tbroyer/gradle-errorprone-plugin#2]
I'm not sure why, but setting up this plugin on a Java 10 project (w/o ErrorProne prior) fails in `compileTestJava`. It does not generate the AutoValue code, but does so in `compileJava`. An `afterEvaluate` shows that the compiler arg `-s, <test dir>` is set, but nothing is generated. The only changes to the build was adding this plugin, so not sure how to diagnose it further.

```gradle
apply plugin: 'net.ltgt.errorprone-javacplugin'

tasks.withType(JavaCompile) {
  options.errorprone {
    disableWarningsInGeneratedCode = true
    excludedPaths = "${buildDir}/generated-sources/.*"
  }
}
```

**Comment by tbroyer** [comment]
I cannot reproduce it. Could you share a repro case?

**Comment by tbroyer** [comment]
Fwiw, here's what I tried:
```gradle
plugins {
  id 'java'
  id 'net.ltgt.errorprone-javacplugin' version '0.2'
}

repositories {
  mavenCentral()
}

dependencies {
  errorprone('com.google.errorprone:error_prone_core:2.3.1')

  compileOnly 'com.google.auto.value:auto-value-annotations:1.6'
  annotationProcessor 'com.google.auto.value:auto-value:1.6'

  testCompileOnly 'com.google.auto.value:auto-value-annotations:1.6'
  testAnnotationProcessor 'com.google.auto.value:auto-value:1.6'
}

tasks.withType(JavaCompile) {
  options.annotationProcessorGeneratedSourcesDirectory = file("$buildDir/generated-sources/$name")

  options.errorprone {
    disableWarningsInGeneratedCode = true
    excludedPaths = "${buildDir}/generated-sources/.*"
  }
}
```
with simple AutoValue classes in `src/main/java` and `src/test/java`.

I also tried with the `net.ltgt.apt` plugin instead of configuring the `options.annotationProcessorGeneratedSourcesDirectory` manually, and without configuring it at all; also without the `options.errorprone` configuration block.

In every case, I could find `AutoValue_*.class` and `AutoValue_*.java` being generated (I of course made sure to `gradle clean` between each `gradle testClasses`).

This is using version 0.2 of the plugin (released yesterday), Gradle 4.7, and JDK 10:
```
$ JAVA_HOME=/usr/lib/jvm/java-10-jdk/ ./gradlew -version
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/home/tbr/.gradle/wrapper/dists/gradle-4.7-bin/73xux3q4721jgr4pjvqf9o30/gradle-4.7/lib/groovy-all-2.4.12.jar) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release

------------------------------------------------------------
Gradle 4.7
------------------------------------------------------------

Build time:   2018-04-18 09:09:12 UTC
Revision:     b9a962bf70638332300e7f810689cb2febbd4a6c

Groovy:       2.4.12
Ant:          Apache Ant(TM) version 1.9.9 compiled on February 2 2017
JVM:          10.0.1 ("Oracle Corporation" 10.0.1+10)
OS:           Linux 4.16.5-1-ARCH amd64
```

**Comment by tbroyer** [comment]
Also note that `testAnnotationProcessor` does not extend `annotationProcessor`, so you have to add annotation processors to each configuration if you want them in both `compileJava` and `compileTestJava`.
